TECH-1129 Add ReactJS administration archetype

sergehuber opened the pull request:
## JIRA

https://jira.jahia.org/browse/TECH-1129

## Description

Adds a new Maven Archetypes to create administration pages with ReactJS. This basically creates a module that displays a "Hello $user" in a new administration page

dgriffon reviewed (changes requested):
- The project needs a README
- It would be good to add some comment on what to fill next once a project created with this archetype.

sergehuber commented:
Ok I have added a global README that also explains what the sub projects are (to avoid creating multiple READMEs) and added a reference to the Academy. It is generic for the moment because the documentation for these archetypes hasn't been written yet (in will hopefully be done soon). 
As for the CI/CD as discussed in our meeting, we will do this separately in another ticket.
